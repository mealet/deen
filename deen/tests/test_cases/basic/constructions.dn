fn main() i32 {
  // IF ELSE construction

  if true {
    println!("if true");
  }

  if true {
    println!("if true else");
  } else {
    println!("wrong");
  }

  if false {
    println!("wrong");
  }

  if false {
    println!("wrong");
  } else {
    println!("if false else")
  }

  // WHILE construction

  println!("countering");
  let counter = 0;
  while counter < 5 {
    println!("{}", counter);
    counter += 1;
  }

  // FOR construction
  
  println!("for countering");
  for counter = 5 {
    println!("{}", counter);
  }

  return 0;
}
