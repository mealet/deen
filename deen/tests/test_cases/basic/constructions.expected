if true
if true else
if false else
countering
0
1
2
3
for countering
0
1
2
3
4
